src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/chips/ChipsRequest.java

```diff
@@ -23,11 +52,51 @@ public String getCompanyNumber() {
         return companyNumber;
     }
 
+    public Map<String, String> getAttachments() {
+        return attachments;
+    }
+
+    public String getReferenceNumber() {
+        return referenceNumber;
+    }
+
+    public String getCustomerEmail() {
+        return customerEmail;
+    }
+
+    public String getReason() {
+        return reason;
+    }
+
+    private Map<String, String> buildAttachmentsMap(String downloadPrefix,
+                                                    List<Attachment> attachments) {
+        Map<String, String> attachmentsMap = new HashMap<>();
+        for (Attachment attachment : attachments) {
+            String name = attachment.getName();
+            Links links = attachment.getLinks();
+            if(links != null) {
+                String downloadLink = String.format("%s%s", downloadPrefix,
+                        links.getLink(ObjectionLinkKeys.DOWNLOAD));
+                attachmentsMap.put(name, downloadLink);
+            }
+        }
+        return attachmentsMap;
+    }
+
     @Override
     public String toString() {
+        StringBuilder sb = new StringBuilder();
+        for(String key : attachments.keySet()) {
+            sb.append(String.format("%s:%s,", key, attachments.get(key)));
+        }
+        String attachments = sb.toString().substring(0, sb.length()-1);
```

Comment:
you can just use String attachments = sb.toString();
substring(0, sb.length()-1) is the whole string.

Reply:
You're right should be -2 as we want to remove the extra trailing comma.